add blog palfish-vldb-paper (#426)
pingcap/blog: making-htap-database-reality-what-i-learned-from-pingcap-vldb-paper.md

---

> title: Making an HTAP Database a Reality: What I Learned from PingCAP's VLDB Paper

Author: ```suggestion
title: "Making an HTAP Database a Reality: What I Learned from PingCAP's VLDB Paper"
```

> title: 'Making an HTAP Database a Reality: What I Learned from PingCAP's VLDB Paper'

Author: ```suggestion
title: Making an HTAP Database a Reality: What I Learned from PingCAP's VLDB Paper
```

> title: "Making an HTAP Database a Reality: What I Learned from PingCAP's VLDB Paper"

Author: ```suggestion
title: [Making an HTAP Database a Reality: What I Learned from PingCAP's VLDB Paper]
```

> title: [Making an HTAP Database a Reality: What I Learned from PingCAP's VLDB Paper]
> author: ['Xianlin Chen']
> date: 2020-10-22
> summary: 'Recently, VLDB published the PingCAP paper, TiDB: A Raft-based HTAP Database. This article, a DBA at PalFish shares his thoughts on the article and his expectations for TiDB's future development.'
> tags: ['HTAP', 'VLDB', 'Consistency', 'Raft']
> categories: ['HTAP']
> image: /images/blog/making-htap-database-reality-what-i-learned-from-pingcap-vldb-paper.jpg
> ---

Author: ```suggestion
title: Making an HTAP Database a Reality: What I Learned from PingCAP's VLDB Paper
author: ['Xianlin Chen']
date: 2020-10-22
summary: Recently, VLDB published the PingCAP paper, TiDB: A Raft-based HTAP Database. This article, a DBA at PalFish shares his thoughts on the article and his expectations for TiDB's future development.
tags: ['HTAP', 'VLDB', 'Consistency', 'Raft']
categories: ['HTAP']
image: /images/blog/making-htap-database-reality-what-i-learned-from-pingcap-vldb-paper.jpg
---
```

**Author:** Xianlin Chen (DBA at PalFish)

![HTAP database](media/making-htap-database-reality-what-i-learned-from-pingcap-vldb-paper.jpg)

> Recently, [VLDB 2020](https://vldb2020.org/) published [PingCAP](https://pingcap.com/)'s paper, [TiDB: A Raft-based HTAP Database](https://en.pingcap.com/blog/vldb-2020-tidb-a-raft-based-htap-database). This is the first paper in the industry to describe the implementation of a distributed [Hybrid Transactional/Analytical Processing](https://en.wikipedia.org/wiki/Hybrid_transactional/analytical_processing) (HTAP) database. As a DBA who benefits greatly from [TiDB](https://docs.pingcap.com/tidb/stable/overview), an open-source, distributed SQL database, I'm happy that VLDB recognized TiDB, and I'm inspired by the PingCAP engineering team's novel ideas.

Author: ```suggestion
Recently, [VLDB 2020](https://vldb2020.org/) published [PingCAP](https://pingcap.com/)'s paper, [TiDB: A Raft-based HTAP Database](https://pingcap.com/blog/vldb-2020-tidb-a-raft-based-htap-database). This is the first paper in the industry to describe the implementation of a distributed [Hybrid Transactional/Analytical Processing](https://en.wikipedia.org/wiki/Hybrid_transactional/analytical_processing) (HTAP) database. As a DBA who benefits greatly from [TiDB](https://docs.pingcap.com/tidb/stable/overview), an open-source, distributed SQL database, I'm happy that VLDB recognized TiDB, and I'm inspired by the PingCAP engineering team's novel ideas.
```

---

> title: 'Making an HTAP Database a Reality: What I learned from PingCAP’s VLDB paper'

Reviewer: ```suggestion
title: 'Making an HTAP Database a Reality: What I Learned from PingCAP's VLDB Paper'
```

Reviewer: Please update all quotes to straight quotes.

author: ['Xianlin Chen']
date: 2020-10-22
summary: 'Recently, VLDB published the PingCAP paper, TiDB: A Raft-based HTAP Database. This article, a DBA at PalFish shares his thoughts on the article and his expectations for TiDB’s future development.'
tags: ['HTAP', 'VLDB', 'Consistency', 'Raft']
categories: ['HTAP']
image: /images/blog/making-htap-database-reality-what-i-learned-from-pingcap-vldb-paper.jpg
---

**Author:** Xianlin Chen (DBA at PalFish)

> ![banner](media/making-htap-database-reality-what-i-learned-from-pingcap-vldb-paper.jpg)

Reviewer: ```suggestion
![HTAP database](media/making-htap-database-reality-what-i-learned-from-pingcap-vldb-paper.jpg)
```

Reviewer: Next time, you can use a shorter image name with SEO keywords.

Reviewer: Descriptive ALT text with an SEO keyword helps our SEO performance, so I want to use "HTAP database."

> Recently, [VLDB 2020](https://vldb2020.org/) published [PingCAP](https://pingcap.com/)’s paper, [TiDB: A Raft-based HTAP Database](https://en.pingcap.com/blog/vldb-2020-tidb-a-raft-based-htap-database). This is the first paper in the industry to describe the implementation of a distributed [Hybrid Transactional/Analytical Processing](https://en.wikipedia.org/wiki/Hybrid_transactional/analytical_processing) (HTAP) database. As a DBA who benefits greatly from [TiDB](https://docs.pingcap.com/tidb/stable/overview), an open-source, distributed SQL database, I’m happy that VLDB recognized TiDB, and I’m inspired by the PingCAP engineering team’s novel ideas.

Reviewer: ```suggestion
Recently, [VLDB 2020](https://vldb2020.org/) published [PingCAP](https://pingcap.com/)’s paper, [TiDB: A Raft-based HTAP Database](https://pingcap.com/blog/vldb-2020-tidb-a-raft-based-htap-database). This is the first paper in the industry to describe the implementation of a distributed [Hybrid Transactional/Analytical Processing](https://en.wikipedia.org/wiki/Hybrid_transactional/analytical_processing) (HTAP) database. As a DBA who benefits greatly from [TiDB](https://docs.pingcap.com/tidb/stable/overview), an open-source, distributed SQL database, I’m happy that VLDB recognized TiDB, and I’m inspired by the PingCAP engineering team’s novel ideas.
```

In this article, I’ll share with you my thoughts about TiDB’s implementation of an HTAP database that provides strong data consistency and resource isolation, as well as my expectations of TiDB in the future.

## The trouble with OLTP and OLAP databases

As you probably know, databases are divided into two types: [Online Transactional Processing](https://en.wikipedia.org/wiki/Online_transaction_processing) (OLTP) and [Online Analytical Processing](https://en.wikipedia.org/wiki/Online_analytical_processing) (OLAP). But have you ever wondered why?

OLTP and OLAP describe two very different data processing methods, and, therefore, they have different database requirements.

| **Characteristics** | **OLTP** | **OLAP** |
|:--|--|--|
| Data queried in one request | Small (a few rows) | Large | 
| Real-time update | Yes | No |
| Transactions | Yes | No | 
| Concurrency | High | Low |
| Query pattern | Similar | Varies a lot |

> <div class="caption-center">Comparing OLTP and OLAP workloads</div>

Reviewer: Generally, we don't add table captions in our posts.

> 
> <div class="caption-center">Comparing OLTP and OLAP workloads</div>

Author: ```suggestion
```
